### ISSUE TITLE
Error in rule prep_pgs_lassosum_i

### ISSUE THREAD

**Issue by marinie** [opain/GenoPred#124]
Hi , I am getting the following error. I was attempting to use a different GWAS sumstat with the test target data and it keeps getting stuck on this error. 

Error in rule prep_pgs_lassosum_i:
    jobid: 52
    input: mjea4/mjea4outdir/reference/gwas_sumstat/EA4/EA4-cleaned.gz, resources/software/install_lassosum.done
    output: mjea4/mjea4outdir/reference/pgs_score_files/lassosum/EA4/ref-EA4.score.gz
    log: mjea4/mjea4outdir/reference/logs/prep_pgs_lassosum_i-EA4.log (check log file(s) for error details)
    conda-env: /home/cytosin3/GenoPred/pipeline/.snakemake/conda/278bb00bb4e2f6526dfc1d09d267a506_
    shell:
        Rscript ../Scripts/pgs_methods/lassosum.R      --ref_plink_chr resources/data/ref/ref.chr      --ref_keep resources/data/ref/keep_files/EUR.keep      --gwas_pop EUR      --sumstats mjea4/mjea4outdir/reference/gwas_sumstat/EA4/EA4-cleaned.gz      --output mjea4/mjea4outdir/reference/pgs_score_files/lassosum/EA4/ref-EA4       --n_cores 1      --pop_data resources/data/ref/ref.pop.txt      --test NA > mjea4/mjea4outdir/reference/logs/prep_pgs_lassosum_i-EA4.log 2>&1
        (one of the commands exited with non-zero exit code; note that snakemake uses bash strict mode!)

Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
Complete log: .snakemake/log/2024-10-26T180407.780777.snakemake.log

The log file referenced above is attached: 
[prep_pgs_lassosum_i-EA4.log](https://github.com/user-attachments/files/17532161/prep_pgs_lassosum_i-EA4.log)

Thank you!
MJ

**Comment by opain** [comment]
Hello MJ, 

Thank you for reaching out. I guess there is something going wrong with the sumstat QC, but it is for reason only throwing an error when it gets to this step. To get a clearer picture of the problem, could you send me the following additional log files:

mjea4/mjea4outdir/reference/pgs_score_files/lassosum/EA4/ref-EA4.log
mjea4/mjea4outdir/reference/gwas_sumstat/EA4/EA4-cleaned.log

This will probably indicate that there are no SNPs in the cleaned GWAS sumstats, or something peculiar, but hopefully the log file from the sumstat QC step will tell us what is going wrong.

Ollie

**Comment by marinie** [comment]
Hi thank you so much for the quick response!
Attached are the files as requested. 
-MJ 

[EA4-cleaned.log](https://github.com/user-attachments/files/17535201/EA4-cleaned.log)
[ref-EA4.log](https://github.com/user-attachments/files/17535202/ref-EA4.log)

**Comment by opain** [comment]
Thank you. The log file for the lassosum script says there are no variants with complete data. Can you send a screenshot of the first few lines of GWAS sumstats (original and cleaned ideally)? The QC script has clearly missed something important.

**Comment by marinie** [comment]
GWAS Sumstats original 
![image](https://github.com/user-attachments/assets/40cd0856-d5f2-4488-84b3-fe4a591f4c2e)
Cleaned
![image](https://github.com/user-attachments/assets/8321202b-feb8-49f8-a52e-872127b319e7)

**Comment by opain** [comment]
Great. Thank you. I can see the issue now. The N column is NA for all rows in the cleaned sumstats. I can see in the log for the sumstat QC that you are specifying the sample size in the gwas_list, which is fine, but there are commas in the value for N. Remove the commas and it will work. I.e change '3,037,499' to '3037499'.

I will make a note to update GenoPred to allow for this as I imagine it happen quite often. Thanks for the report.

Ollie

**Comment by marinie** [comment]
Hi Ollie, we corrected that issue and now have run into a different one 

Error in rule indiv_report_i:
    jobid: 56
    input: resources/software/install_ggchicklet.done, mjea4/mjea4outdir/reference/pgs_score_files/lassosum/EA4/ref-EA4.score.gz, mjea4/mjea4outdir/reference/target_checks/example_plink1/target_pgs.done, mjea4/mjea4outdir/reference/gwas_sumstat/EA4/EA4-cleaned.gz
    output: mjea4/mjea4outdir/reference/target_checks/example_plink1/indiv_report-6_AMR.6_AMR-report.done
    log: mjea4/mjea4outdir/reference/logs/indiv_report_i-example_plink1-6_AMR.6_AMR.log (check log file(s) for error details)
    conda-env: /home/cytosin3/GenoPred/pipeline/.snakemake/conda/278bb00bb4e2f6526dfc1d09d267a506_
    shell:
        
    {
      mkdir -p resources/tmp/example_plink1.6_AMR.6_AMR/ ;       cp ../Scripts/pipeline_reports/indiv_report_creator.Rmd resources/tmp/example_plink1.6_AMR.6_AMR//indiv_report_creator.Rmd ;       mkdir -p mjea4/mjea4outdir/example_plink1/reports/individual;       Rscript -e "rmarkdown::render('resources/tmp/example_plink1.6_AMR.6_AMR//indiv_report_creator.Rmd',       output_file = '/home/cytosin3/GenoPred/pipeline/mjea4/mjea4outdir/example_plink1/reports/individual/example_plink1-6_AMR.6_AMR-report.html',       params = list(name = 'example_plink1', id = '6_AMR.6_AMR', config = 'mjea4/mjea4config.yaml', cwd = '/home/cytosin3/GenoPred/pipeline'))" ;       rm -r resources/tmp/example_plink1.6_AMR.6_AMR/
    } > mjea4/mjea4outdir/reference/logs/indiv_report_i-example_plink1-6_AMR.6_AMR.log 2>&1


[indiv_report_i-example_plink1-6_AMR.6_AMR.log](https://github.com/user-attachments/files/17544422/indiv_report_i-example_plink1-6_AMR.6_AMR.log)

**Comment by opain** [comment]
No problem. This is likely because you are using GWAS sumstats that only include top SNPs (only a few thousand variants are present) - The pipeline uses lassosum to estimate the R2 of the PGS, which is in turn used to calibrate the PGS. The error you got indicates the R2 estimate is NA. I will update to avoid the script failing. 

Is the individual-level report something you actually want? If not you could just set the indiv_report to F in the target_list. I would recommend doing this for large samples of data anyway as it takes a long time to run and isn't usually useful. 

You can find an example individual-level report here: https://opain.github.io/GenoPred/example_plink1-1_EUR.1_EUR-report.html

Eve if you turn this off, you will still get a sample-level report which usually more useful. Example: https://opain.github.io/GenoPred/example_plink1-report.html

**Comment by marinie** [comment]
Hi Ollie, 

Thank you! What I need is the individual-level polygenic scores for each individual in my target sample. What  I am aiming to do is to use those scores in a regression analyses with different outcome variables. Since I only need the one polygenic score per individual, is there a way for me to read those scores into R without generating the entire individual-level report? Would that be more feasible?

Thanks!
-MJ

**Comment by opain** [comment]
Yes, the PGS for your target samples will already have been completed (it is required for the individual-level report to run). The target sample PGS could have been requested from the pipeline using the target_pgs rule (commands to request specific outputs can be found here https://opain.github.io/GenoPred/pipeline_readme.html#Requesting_outputs).

You can find the PGS files in the folder <outdir>/<target_pgs>/<pgs> in separate folders for each population, method and GWAS.

I have created some R functions to facilitate the process of reading them into R. See the read_pgs function: https://opain.github.io/GenoPred/pipeline_readme.html#Reading_outputs_into_R

There is an example of me using it in the 'Evaluate PGS' section of this page https://opain.github.io/GenoPred/opensnp_benchmark.html#GenoPred_V2

I hope this helps.
